apps/site/pages/en/learn/asynchronous-work/promises-in-nodejs.md

```diff
@@ -0,0 +1,264 @@
+- **Rejected**: The operation failed, and the Promise is settled with a reason (usually an error).
+
+For example, imagine you're waiting for an email from a friend. You're in the pending state. If your friend replies, you've entered the fulfilled state. If the email bounces back, you are in the rejected state.
+
+## Basic Syntax of a Promise
+
+A Promise is created using the `new Promise()` constructor. The constructor takes a function with two parameters: `resolve` and `reject`. These functions are used to transition the Promise from the **pending** state to either **fulfilled** or **rejected**.
+
+```js
+const myPromise = new Promise((resolve, reject) => {
+  let success = true;
+
+  if (success) {
+    resolve('Operation was successful!');
+  } else {
+    reject('Something went wrong.');
+  }
+});
+```
+
+In the above example:
+
+- If the `success` condition is `true`, the Promise is fulfilled and the value `'Operation was successful!'` is passed to the `resolve` function.
+- If the `success` condition is `false`, the Promise is rejected and the error `'Something went wrong.'` is passed to the `reject` function.
+
+## Handling Promises with `.then()` and `.catch()`
+
+Once a Promise is created, you can handle the outcome by using the `.then()` and `.catch()` methods.
+
+- `.then()` is used to handle a fulfilled Promise and access its result.
+- `.catch()` is used to handle a rejected Promise and catch any errors that may occur.
+
+```js
+const myPromise = new Promise((resolve, reject) => {
+  let success = true;
+
+  if (success) {
+    resolve('Operation was successful!');
+  } else {
+    reject('Something went wrong.');
+  }
+});
+
+myPromise
+  .then(result => {
+    console.log(result); // This will run if the Promise is fulfilled
+  })
+  .catch(error => {
+    console.error(error); // This will run if the Promise is rejected
+  });
+```
+
+## Chaining Promises
+
+One of the great features of Promises is that they allow you to chain multiple asynchronous operations together. When you chain Promises, each `.then()` block waits for the previous one to complete before it runs.
+
+```js
+const promise1 = new Promise((resolve, reject) => {
+  setTimeout(() => resolve('First task completed'), 1000);
+});
+
+const promise2 = new Promise((resolve, reject) => {
+  setTimeout(() => resolve('Second task completed'), 1000);
+});
+
+promise1
+  .then(result => {
+    console.log(result); // 'First task completed'
+    return promise2; // Return the second Promise
+  })
+  .then(result => {
+    console.log(result); // 'Second task completed'
+  })
+  .catch(error => {
+    console.error(error); // If any Promise is rejected, catch the error
+  });
+```
+
+In this example:
+
+- The first Promise resolves after 1 second, and its result is logged.
+- The second Promise resolves after another second, and its result is logged after the first one finishes.
+
+## Using Async/Await with Promises
```

Comment:
This whole paragraph is a bit too simplistic. async/await is a bit more complex than that.

You can refer to the talk/workshop @jasnell did when at NearForm: https://www.youtube.com/watch?v=XV-u_Ow47s0. I would recommend adding it as a reference, too.

Reply:
@jasnell I've added your talk as a reference, just confirming that [we have permission](https://github.com/nodejs/nodejs.org/issues/7200#issuecomment-2564402143) to do so?

Reply:
No need to ask. Feel free to use any of my talks/write ups.